spec/message_spec.rb

```diff
@@ -4,7 +4,7 @@
   let(:mock_channel) { instance_double("Channel") }
   let(:delivery_info) { instance_double("DeliveryInfo", :channel => mock_channel, :delivery_tag => "a_tag") }
   let(:headers) { instance_double("Headers") }
-  let(:message) { Message.new(delivery_info, headers, { "a" => "payload" }) }
+  let(:message) { Message.new({ "a" => "payload" }, headers, delivery_info) }
```

Comment:
`headers` isn't used at all in the test, so you can inline that too so we have less `let`s.